### PR TITLE
scheduler: fix region scatter may transfer leader to removed peer

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
Somtimes region scatter may timeout, it cause by:
- Leader already changed but scatter-region step still use old leader
- Leader may transfer to a removed peer

log like:
```
[region 2109921] operator timeout: scatter-region (kind:leader,region,admin, region:2109921(5233,677), createAt:2019-03-26 14:29:14.885583251 +0800 CST m=+1888726.900848985, currentStep:10, steps:[add learner peer 2109925 on store 5 promote learner peer 2109925 on store 5 to voter remove peer on store 8 transfer leader from store 0 to store 5 add learner peer 2109926 on store 6 promote learner peer 2109926 on store 6 to voter remove peer on store 4 transfer leader from store 0 to store 6 add learner peer 2109927 on store 1 promote learner peer 2109927 on store 1 to voter transfer leader from store 7 to store 8 remove peer on store 7 transfer leader from store 0 to store 1]) timeout

add  store 5
remove store 8
transfer to 5
add store 6
remove store 4
transfer to 6
add  store 1 
transfer from 7 to 8
remove  store 7
transfer to 1
```
the origin leader is on store 7.
### What is changed and how it works?
Fix the operator create logic.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by disksing** [comment]
@nolouch The log you provided does not transfer leader to removed peer. The steps are:
```
- add 5
- transfer 2
- remove 1
- transfer 5
- add 4
- remove 2
- transfer 4
```

**Comment by nolouch** [comment]
@disksing  I update the log, this one is pick from test cluster.

**Comment by rleungx** [comment]
@nolouch CI failed.

**Comment by nolouch** [comment]
PTAL @disksing @rleungx

File: server/schedule/region_scatterer.go [hunk 1]
```diff
@@ -105,31 +111,90 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo) *Operator {
...
+func (r *RegionScatterer) createOperator(origin *core.RegionInfo, replacedPeers, targetPeers []*metapb.Peer) *Operator {
+	// Randomly pick a leader
+	i := rand.Intn(len(targetPeers))
+	targetLeaderPeer := targetPeers[i]
+
+	storeIDs := origin.GetStoreIds()
+	steps := make([]OperatorStep, 0, len(targetPeers)*2+1)
+	deferSteps := make([]OperatorStep, 0, 2)
+	var kind OperatorKind
+	sameLeader := targetLeaderPeer.GetStoreId() == origin.GetLeader().GetStoreId()
```
Comment (rleungx) [inline server/schedule/region_scatterer.go]: How about using a variable for `origin.GetLeader().GetStoreId()`

File: server/schedule/region_scatterer.go [hunk 2]
```diff
@@ -105,31 +111,90 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo) *Operator {
...
-		if err != nil {
-			continue
+func (r *RegionScatterer) createOperator(origin *core.RegionInfo, replacedPeers, targetPeers []*metapb.Peer) *Operator {
+	// Randomly pick a leader
+	i := rand.Intn(len(targetPeers))
+	targetLeaderPeer := targetPeers[i]
+
+	storeIDs := origin.GetStoreIds()
+	steps := make([]OperatorStep, 0, len(targetPeers)*2+1)
+	deferSteps := make([]OperatorStep, 0, 2)
```
Comment (rleungx) [inline server/schedule/region_scatterer.go]: It's better to add some comments to explain why we put these steps in the end.

**Pull request merged**